lib/puppet/file_system/file_impl.rb

```diff
@@ -156,12 +156,14 @@ def chmod(mode, path)
   end
 
   def replace_file(path, mode = nil)
-    mode ||= begin
-               stat = Puppet::FileSystem.lstat(path)
-               stat.mode & 07777
-             rescue Errno::ENOENT
-               0640
-             end
+    begin
+      stat = Puppet::FileSystem.lstat(path)
+      gid = stat.gid
+      uid = stat.uid
+      mode ||= stat.mode & 07777
+    rescue Errno::ENOENT
+      mode ||= 0640
+    end
```

Comment:
The `jruby` file implementation subclasses `posix`. Does the jruby implementation need any updates? It looks like `windows` already handles owner and group due to `get_dacl_from_file`.

Reply:
It will also work since `jruby` `replace_file` just checks if `path` is a directory and than calls `super`